### ISSUE TITLE
Incorrect __module__ attribute for _struct.Struct and perhaps a few others

### ISSUE THREAD

**Issue by df79943f-4aee-4531-a00d-c6b12816eb70** [python/cpython#79512]
BPO | [35331](https://bugs.python.org/issue35331)
--- | :---
Nosy | @mdickinson, @meadori, @serhiy-storchaka, @mr-nfamous

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2018-11-27.18:22:25.306>
labels = ['3.8', 'type-bug', '3.7']
title = 'Incorrect __module__ attribute for _struct.Struct and perhaps a few others'
updated_at = <Date 2019-05-23.12:16:40.753>
user = 'https://github.com/mr-nfamous'
```

bugs.python.org fields:
```python
activity = <Date 2019-05-23.12:16:40.753>
actor = 'bup'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = []
creation = <Date 2018-11-27.18:22:25.306>
creator = 'bup'
dependencies = []
files = []
hgrepos = []
issue_num = 35331
keywords = []
message_count = 1.0
messages = ['330544']
nosy_count = 4.0
nosy_names = ['mark.dickinson', 'meador.inge', 'serhiy.storchaka', 'bup']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue35331'
versions = ['Python 3.7', 'Python 3.8']
```

</p></details>

**Comment by serhiy-storchaka** [comment]
It was fixed for `_struct` in 4f384af067d05b16a554bfd976934fca9f87a1cf (bpo-38076) and for `_contextvars` in 988f1ec8d2643a0d00851903abcdae90d57ac0e6 (bpo-1635741).

hamt, items, etc are private types.

**Comment by mscuthbert** [comment]
Hi all -- Thanks for the fix.  I wish though that this were done on a major version upgrade, because I did not think that `Class.__name__` was considered private (at least it's documented in the data model) and it ended up making 3.10.6 a backwards incompatible version in some places where .__name__ was being used instead of isinstance() checking (long old code from the time when isinstance() checking was very slow, but the code still worked)
